### DISCUSSION TITLE
상속한 부모클래스를 GetComponent 로 호출

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#1197] [5. Q&A]
ZepetoWorld` 1.6.0 입니다. 

상속한 부모클래스를 GetComponent 로 호출할 수 있나요?

class A : Unit {
     key: string = "A"
}

class B : Unit {
     key: string = "B"
}

--------------------------------------
const unit = `GetComponent<Unit>();` <- 이게 가능한가요? 원래 가능해야 하는데, 안되는것 같아서요.
unit.key <- 에러 납니다.  unit 이 null 입니다. 

지원하지 않나요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 말씀 주신 내용은 버그로 판별되어 추후에 수정 될 예정입니다.
정확한 적용 시점은 확답드리기 어려운 점 양해 부탁 드립니다.
감사합니다.

**Comment by peterkangdev** [comment]
모듈화를 하고, 추후 확장하려면 상속이 필수인데... 조속히 업데이트 되면 좋겠습니다. 
if 나 case 로 분기하려니 소스가 지저분해지고, 너무 관리하기 힘드네요.. ㅠㅠ
